MaxText/deployment.yaml

```diff
@@ -0,0 +1,186 @@
+apiVersion: v1
```

Comment:
@MattIrv @divrawal I updated the yaml file to better match what we will provide in the google3 doc and removed a couple of hacks as we are having the correct GCSFuse CSI Driver version now. Please take a look